src/server/modes/charts/plugins/ql/js.ts

```diff
@@ -246,6 +252,24 @@ export default ({shared, ChartEditor}: {shared: QlConfig; ChartEditor: IChartEdi
             });
         }
 
+        if (newVisualization.placeholders[0]?.items[0]) {
+            if (
+                !applyDefaultSorting &&
+                (LINEAR_VISUALIZATIONS.has(newVisualization.id) ||
+                    newVisualization.id === WizardVisualizationId.BarXD3)
+            ) {
+                newVisualization.placeholders[0].settings = {
+                    axisModeMap: {
+                        [newVisualization.placeholders[0].items[0].guid]: 'discrete',
+                    },
+                };
+
+                newVisualization.placeholders[0].items[0].disableAxisMode = true;
```

Comment:
`disableAxisMode`
Why is this a characteristic of the field? Perhaps you should specify this in the X-axis placeholder settings?